### PR TITLE
public profile for 2018-12

### PR DESCRIPTION
**Pull request by sijuman**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: profile/2018-12-01-profile.json [hunk 1]
```diff
@@ -0,0 +1,623 @@
...
+        },
+
+        "microsoft.network": {
+
+                         "2018-05-01":[
+                           "dnszones",
+                           "dnsOperationResults",
+                           "dnsOperationStatuses",
+                           "getDnsResourceReference",
+                           "internalNotify",
```
Comment (anuchandy) [inline profile/2018-12-01-profile.json]: `dnsOperationResults`, `dnsOperationStatuses`, `internalNotify` endpoints does not exists in microsoft.network `2018-05-01`, see [here](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/dns/resource-manager/Microsoft.Network/stable/2018-05-01/dns.json).

File: profile/2018-12-01-profile.json [hunk 2]
```diff
@@ -0,0 +1,623 @@
...
+                             "virtualNetworkGateways",
+                             "virtualNetworks",
+                             "virtualNetworkTaps"
+                         ]
+        },
+
+        "microsoft.notificationHubs": {
+
+                         "2016-03-01": [
+                             "operationResults",
```
Comment (anuchandy) [inline profile/2018-12-01-profile.json]: `operationResults` endpoint is not defined in microsoft.notificationHubs `2016-03-01` see [here](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/stable/2016-03-01/notificationhubs.json).

File: profile/2018-12-01-profile.json [hunk 3]
```diff
@@ -0,0 +1,623 @@
...
+                            "workspaceCollections"
+                        ]
+        },
+
+        "microsoft.powerBIDedicated": {
+                        "2017-10-01":[
+                            "capacities",
+                            "locations/checkNameAvailability",
+                            "locations/operationresults",
+                            "locations/operationstatuses"
```
Comment (anuchandy) [inline profile/2018-12-01-profile.json]: `locations/operationresults` and `locations/operationstatuses` endpoint does not exists in microsoft.powerBIDedicated `2017-10-01` see [here](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/powerbidedicated/resource-manager/Microsoft.PowerBIdedicated/stable/2017-10-01/powerbidedicated.json).

**Review by anuchandy — CHANGES_REQUESTED** [review]
@sijuman Many of the resources specified in the profile does not exists. I've pointed around 20+ of them but seems there are more.  Could you please go through entire profile and make sure entries are pointing to existing api-version and resources exists in the spec repo? This is very important to fix before we generate and publish any SDKs.

File: profile/2018-12-01-profile.json [hunk 4]
```diff
@@ -0,0 +1,623 @@
...
+                        ]
+        },
+
+        "microsoft.authorization": {
+
+                        "2015-06-01": [
+                            "classicAdministrators"
+                        ],
+
+                        "2017-09-01":[
```
[the remark this answers is not in the document]
Comment (lmazuel) [reply profile/2018-12-01-profile.json]: Nor here https://github.com/Azure/azure-rest-api-specs/tree/master/specification/resources/resource-manager/Microsoft.Authorization/stable

File: profile/2018-12-01-profile.json [hunk 5]
```diff
@@ -0,0 +1,623 @@
...
+                            "roleAssignments"
+                        ],
+
+                        "2018-03-01":[
```
[the remark this answers is not in the document]
Comment (sijuman) [reply profile/2018-12-01-profile.json]: These exist....check https://github.com/Azure/azure-rest-api-specs/tree/master/specification/resources/resource-manager/Microsoft.Authorization/stable/2018-03-01

File: profile/2018-12-01-profile.json [hunk 6]
```diff
@@ -0,0 +1,623 @@
...
+                        ],
+                        "2017-04-01": [
+                            "checkNameAvailability",
+                            "namespaces",
+                            "operations",
+                            "sku"
+                        ]
+        },
+
+        "microsoft.features": {
```
[the remark this answers is not in the document]
Comment (sijuman) [reply profile/2018-12-01-profile.json]: Check underneath Microsoft.Resources folder

**Comment by salameer** [comment]
Hi @anuchandy what's the status on this, it's been open for a while now

**Comment by anuchandy** [comment]
@salameer we had email conversation with profile team on this long back, the issue is - many of the model and api declared in this swagger as supported by an api-version is not exists in the spec for that api Version. I haven't heard anything from profile team on this. 

@sijuman closing this PR due to no action. \\cc @joshgav

**Pull request closed without merging**